### PR TITLE
Added Bluemix app template

### PR DESCRIPTION
**Pull request by hacksparrow**
### Description

Adds Bluemix app template files.

#### Related issues

- None

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] Add basic files
- [ ] Convert an existing LB project to a BM  project
- [x] Add bluemix-autoscaling-agent 
- [x] Add appmetrics 
- [x] Add Bluemix services transparently
- [x] Add service discovery
  - [x] Cloudant
  - [x] MongoDB
  - [ ] Redis (confirmation blocked by Bluemix' limitation)
  - [ ] PostgreSQL (confirmation blocked by Bluemix' limitation)
  - [ ] MySQL(confirmation blocked by Bluemix' limitation)
- [x] Docker files
  - [x] `Dockerfile`
  - [x] `Dockerfile-run` (not required)
  - [x] `Dockerfile-tools` (not required)
- [x] Ensure `Dockerfile` is applied
- [x] Add "Create Toolchain" button
- [x] Make `manifest.yml` configurable
- [x] Ensure `manifest.yml` is applied
- [x] Make other generated files configurable

### CODE REVIEW

File: templates/bluemix/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+declared-services:
```
Comment (raymondfeng) [inline templates/bluemix/manifest.yml]: This part should be optional. The user can opt in to provisioning BM services.

File: templates/bluemix/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  cloudant-db:
+    label: cloudantNoSQLDB
+    plan: Lite
+applications:
+- path: .
+  memory: 256M
```
Comment (raymondfeng) [inline templates/bluemix/manifest.yml]: Do we prompt the user to override some of the defaults from the CLI?

Comment (hacksparrow) [inline templates/bluemix/manifest.yml]: After detecting `--bluemix`, we should prompt the users for values for all overridable Bluemix defaults.

@seabaylea you might want to join us in the discussion.

File: templates/bluemix/manifest.yml (hunk 1 above)
Comment (hacksparrow) [inline templates/bluemix/manifest.yml]: I will keep only the bare basics for now, and remove it. Let's develop the interface and content of the file progressively.

File: templates/bluemix/manifest.yml (hunk 2 above)
Comment (seabaylea) [inline templates/bluemix/manifest.yml]: @raymondfeng @hacksparrow for not I'd just hard-code memory the a reasonable value. A user can always modify it manually afterwards.

File: templates/bluemix/bluemix/toolchain.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+name: Continuous Delivery Toolchain
+description: "This toolchain includes tools to develop and deploy your app. Depending on your app, when you create the toolchain, the GitHub repository will either be empty or will contain source code from your app.\n\nThis toolchain uses tools that are part of the Continuous Delivery service. If an instance of that service isn't already in your organization, when you click **Create**, it is automatically added at no cost to you. For more information and terms, see the [Bluemix catalog](/catalog/services/continuous-delivery/).\n\nTo get started, click **Create**."
+version: 0.2
+required: 
+ - deploy
+ - repo
+
+toolchain:
+  name: knownapp
```
Comment (seabaylea) [inline templates/bluemix/bluemix/toolchain.yml]: This should be replaced with he tname of the application

File: templates/bluemix/bluemix/toolchain.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+webide:
+  service_id: orion
+  
+#Deployment
+deploy:
+  schema: 
+    $ref: deploy.json
+  service-category: pipeline
+  parameters:
+    app-name: knownapp
```
Comment (seabaylea) [inline templates/bluemix/bluemix/toolchain.yml]: This should be replaced with the name of the application

File: templates/bluemix/manifest.yml (hunk 2 above)
Comment (hacksparrow) [inline templates/bluemix/manifest.yml]: I am developing optional prompts for those values with current values as defaults.

File: templates/bluemix/bluemix/toolchain.yml (hunk 3 above)
Comment (hacksparrow) [inline templates/bluemix/bluemix/toolchain.yml]: Noted.

File: templates/bluemix/bluemix/toolchain.yml (hunk 4 above)
Comment (hacksparrow) [inline templates/bluemix/bluemix/toolchain.yml]: Noted.

**Pull request closed without merging**